### PR TITLE
[Storage] az storage blob upload: Migrate blob upload to Track 2 and make block blob support setting access tier

### PR DESCRIPTION
**Pull request by zhoxing-ms**
Feature request: #13841 #12766

**Description**  

1. Migrate blob upload to Track 2
2. Make the block blob support setting access tier while uploading

**Testing Guide**  

Run `az storage blob upload`

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S173

File: src/azure-cli/azure/cli/command_modules/storage/commands.py [hunk 1]
```diff
@@ -284,8 +286,6 @@ def get_custom_sdk(custom_module, client_factory, resource_type=ResourceType.DAT
                                     'undeleted'),
                                 table_transformer=transform_boolean_for_table, min_api='2017-07-29')
 
-        g.storage_custom_command_oauth('upload', 'upload_blob',
```
Comment (qianwens) [inline src/azure-cli/azure/cli/command_modules/storage/commands.py:287]: >  g.storage_custom_command_oauth('upload', 'upload_blob', [](start = 6, length = 57)

what about Azure stack if you delete this one

Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/storage/commands.py:287]: I keep the original method call of Track 1 in another file `commands_azure_stack.py`.
https://github.com/Azure/azure-cli/blob/4d58aaee1266a46ed320c9cf21e879163f05ba91/src/azure-cli/azure/cli/command_modules/storage/commands_azure_stack.py#L269

**Pull request closed without merging**